### ISSUE TITLE
Enabling/Disabling Ocean Plane (At Times) Causes Crash in 4.19 P1

### ISSUE THREAD

**Issue by reapazor** [UE4-OceanProject/OceanProject#61]
During edit (not even PIE) turning on and off the ocean plane can cause a crash intermittently.
This may be limited to macOS.

Will further investigate, but just though I would dump in here for record.

```
FPlanarReflectionParameters::SetParameters(FRHICommandList&, FRHIPixelShader*, FSceneView const&, FPlanarReflectionSceneProxy const*) Address = 0x11a818e15 (filename not found) [in UE4Editor-Renderer.dylib]
FBasePassReflectionParameters::SetMesh(FRHICommandList&, FRHIPixelShader*, FSceneView const&, FPrimitiveSceneProxy const*, ERHIFeatureLevel::Type) Address = 0x11a48dafb (filename not found) [in UE4Editor-Renderer.dylib]
TBasePassDrawingPolicy<FUniformLightMapPolicy>::SetMeshRenderState(FRHICommandList&, FViewInfo const&, FPrimitiveSceneProxy const*, FMeshBatch const&, int, FDrawingPolicyRenderState&, TBasePassDrawingPolicy<FUniformLightMapPolicy>::ElementDataType const&, FMeshDrawingPolicy::ContextDataType) const Address = 0x11a5265f4 (filename not found) [in UE4Editor-Renderer.dylib]
int TStaticMeshDrawList<TBasePassDrawingPolicy<FUniformLightMapPolicy> >::DrawElement<(InstancedStereoPolicy)2>(FRHICommandList&, FViewInfo const&, FMeshDrawingPolicy::ContextDataType, FDrawingPolicyRenderState&, TStaticMeshDrawList<TBasePassDrawingPolicy<FUniformLightMapPolicy> >::FElement const&, unsigned long long, TStaticMeshDrawList<TBasePassDrawingPolicy<FUniformLightMapPolicy> >::FDrawingPolicyLink*, bool&) Address = 0x11a528315 (filename not found) [in UE4Editor-Renderer.dylib]
bool TStaticMeshDrawList<TBasePassDrawingPolicy<FUniformLightMapPolicy> >::DrawVisibleInner<(InstancedStereoPolicy)2>(FRHICommandList&, FViewInfo const&, FMeshDrawingPolicy::ContextDataType, FDrawingPolicyRenderState&, TBitArray<SceneRenderingBitArrayAllocator> const*, TArray<unsigned long long, TMemStackAllocator<0u> > const*, StereoPair const*, int, int, bool) Address = 0x11a527de6 (filename not found) [in UE4Editor-Renderer.dylib]
FDeferredShadingSceneRenderer::RenderBasePassStaticDataType(FRHICommandList&, FViewInfo&, FDrawingPolicyRenderState const&, EBasePassDrawListType) Address = 0x11a495b1d (filename not found) [in UE4Editor-Renderer.dylib]
FDeferredShadingSceneRenderer::RenderBasePassStaticData(FRHICommandList&, FViewInfo&, FDrawingPolicyRenderState const&) Address = 0x11a498aae (filename not found) [in UE4Editor-Renderer.dylib]
FDeferredShadingSceneRenderer::RenderBasePassView(FRHICommandListImmediate&, FViewInfo&, FExclusiveDepthStencil::Type) Address = 0x11a49586a (filename not found) [in UE4Editor-Renderer.dylib]
FDeferredShadingSceneRenderer::RenderBasePass(FRHICommandListImmediate&, FExclusiveDepthStencil::Type) Address = 0x11a494e28 (filename not found) [in UE4Editor-Renderer.dylib]
FDeferredShadingSceneRenderer::Render(FRHICommandListImmediate&) Address = 0x11a5895de (filename not found) [in UE4Editor-Renderer.dylib]
RenderViewFamily_RenderThread(FRHICommandListImmediate&, FSceneRenderer*) Address = 0x11add44e8 (filename not found) [in UE4Editor-Renderer.dylib]
TGraphTask<FRendererModule::BeginRenderingViewFamily(FCanvas*, FSceneViewFamily*)::EURCMacro_FDrawSceneCommand>::ExecuteTask(TArray<FBaseGraphTask*, FDefaultAllocator>&, ENamedThreads::Type) Address = 0x11adf5e3e (filename not found) [in UE4Editor-Renderer.dylib]
FNamedTaskThread::ProcessTasksNamedThread(int, bool) Address = 0x10a9d7b82 (filename not found) [in UE4Editor-Core.dylib]
FNamedTaskThread::ProcessTasksUntilQuit(int) Address = 0x10a9d63a8 (filename not found) [in UE4Editor-Core.dylib]
FTaskGraphImplementation::ProcessThreadUntilRequestReturn(ENamedThreads::Type) Address = 0x10a9d24ee (filename not found) [in UE4Editor-Core.dylib]
RenderingThreadMain(FEvent*) Address = 0x1107dd655 (filename not found) [in UE4Editor-RenderCore.dylib]
FRenderingThread::Run() Address = 0x1107f38d7 (filename not found) [in UE4Editor-RenderCore.dylib]
FRunnableThreadPThread::Run() Address = 0x10aa4f44e (filename not found) [in UE4Editor-Core.dylib]
FRunnableThreadPThread::_ThreadProc(void*) Address = 0x10a9f8887 (filename not found) [in UE4Editor-Core.dylib]
_pthread_body Address = 0x7fff70fd76c1 (filename not found) [in libsystem_pthread.dylib]
_pthread_body Address = 0x7fff70fd756d (filename not found) [in libsystem_pthread.dylib]
thread_start Address = 0x7fff70fd6c5d (filename not found) [in libsystem_pthread.dylib]
```

**Comment by TheZoc** [comment]
Hello @reapazor,

I have updated your comment so it was easier to read the crash dump.

I'm not sure how debugging works in macOS. Is there an option to download symbols when you install
 UE4? This is how it looks on windows (The option name is "Editor Symbols for debugging").
 
![symbols](https://user-images.githubusercontent.com/924927/35411211-1260bdd0-01ff-11e8-9c8a-ba3325663ee3.png)


Maybe this will help find the source of the issue, if this option is available :)

**Comment by reapazor** [comment]
Hehehe :) That dump was from a rocket build without editor debug ... I'm running 4.19 branch build currently with full debug info ...so if it crashes again I'll update here.
One thing to note is that I'm seeing 'horrific' driver hangs on macOS not connected to the ocean plane, so this in itself could be tied to the 4.19 preview branch.

**Comment by EvoPulseGaming** [comment]
They made major changes to landscape rendering, (particularly around tessellation ) that seems to have caused all sorts of new bugs (but MAJOR tessellation fps increase), so since planes* and landscape seem to be similar (landscape is probably parented or shares functions) I would chalk this up to preview 4.19.
*I mention planes, because the infinate ocean is a plane, I think...

I myself, downgraded back to 4.18 after I found that I was constantly getting crashed in an otherwise fine ocean project.

Oh, they plan on doing reworking the shadows for landscape in 4.20, so expect crashing in 4.20 preview too!

**Comment by EvoPulseGaming** [comment]
We are at 4.19v6 can we get an update if this is still an issue?

**Comment by reapazor** [comment]
I havent checked recently, but it was happening on source pulls. I havent checked on release ver

**Comment by TheZoc** [comment]
Can you check if this was fixed in f3a5f5dc8e6fb1e35901f8f2a0b38446eaf3d728 ? :)

**Comment by TheZoc** [comment]
Closing this issue, as it seems it isn't a problem anymore :)
